libdrgn/arch_aarch64.c

```diff
@@ -68,6 +68,100 @@ static void demangle_cfi_registers_aarch64(struct drgn_program *prog,
 	drgn_register_state_set_from_u64(prog, regs, x30, ra.value);
 }
 
+static struct drgn_error *
+get_initial_registers_from_struct_aarch64(struct drgn_program *prog,
+					  const void *buf, size_t size,
+					  struct drgn_register_state **ret);
+
+static struct drgn_error *
+get_stackframe_offset(struct drgn_program *prog, uint64_t *ret)
+{
+	if (!prog->aarch64_stackframe_offset_cached) {
+		struct drgn_error *err;
+		struct drgn_qualified_type pt_regs_type;
+
+		err = drgn_program_find_type(prog, "struct pt_regs", NULL, &pt_regs_type);
+		if (err)
+			return err;
+
+		struct drgn_type_member *stackframe_member;
+		uint64_t bit_offset;
+		err = drgn_type_find_member(pt_regs_type.type, "stackframe",
+					    &stackframe_member, &bit_offset);
+		if (err)
+			return err;
+
+		prog->aarch64_stackframe_offset_cached = bit_offset / 8;
+	}
+
+	*ret = prog->aarch64_stackframe_offset_cached;
+	return NULL;
+}
+
+
+static struct drgn_error *
+fallback_unwind_try_pt_regs(struct drgn_program *prog, uint64_t fp,
+			    uint64_t frame[2],
+			    struct drgn_register_state *regs,
+			    struct drgn_register_state **ret)
+{
+	struct drgn_error *err;
+	uint64_t stackframe_offset;
+
+	err = get_stackframe_offset(prog, &stackframe_offset);
+	if (err)
+		return err;
+
+	uint64_t pt_regs_addr = fp - stackframe_offset;
+
+	if (!frame[0] && !frame[1]) {
+		// Since c2c6b27b5aa14 ("arm64: stacktrace: unwind exception
+		// boundaries") - v6.13, a signal frame with both values zero
```

Comment:
"Signal frame" confused me here. I think it'd be more accurate to say something like "a frame record with the fp and lr both zero".